woocommerce/woocommerce-ios
# Inbox view part 1: list view with view model and sync states

Part of #5954 


### Description


This is the first step to implementing Inbox UI. Due to PR size, this PR mainly contains a SwiftUI view `Inbox` that has 3 states based on the syncing status:
- `results`: when there are results to be shown
- `empty`: when there are no results
- `syncingFirstPage`: when syncing the first page of inbox notes

I decided to go with a simple state transition following other examples in the codebase (e.g. `AddProductToOrderViewModel`) instead of building a state machine due to `@Published` complexity.

The view just shows an existing `TitleAndSubtitleRow` for each note. In a later PR, there is a new SwiftUI row view and its corresponding view models (note row and action). Reviewers can focus on the state changes in this PR.

### Testing instructions


The UI for each row hasn't been implemented yet, but feel free to check out the list of notes with temporary UI using `TitleAndSubtitleRow`:

- Launch the app
- Go to the Menu tab
- Tap on "Inbox" --> after syncing, a list of notes with title and HTML content are shown

### Screenshots


<img src="https://user-images.githubusercontent.com/1945542/153807959-cafc8a64-aa6d-4b0b-9214-71ec8b66b732.png" width="300" />

---
